pyiceberg/catalog/snowflake_catalog.py

```diff
@@ -0,0 +1,289 @@
+            parts = identifier.split(".")
+            if len(parts) == 1:
+                return cls(None, parts[0], None)
+            elif len(parts) == 2:
+                return cls(parts[0], parts[1], None)
+
+            raise ValueError(f"Invalid identifier: {identifier}")
+
+        @property
+        def table_name(self) -> str:
+            return ".".join(self)
+
+        @property
+        def schema_name(self) -> str:
+            return ".".join(self)
+
+    def __init__(self, name: str, **properties: str):
+        """
+        params:
+            name: Name of the catalog.
+            user: Snowflake user.
+            account: Snowflake account.
+            authenticator: Snowflake authenticator.
+            password: Snowflake password.
+            private_key: Snowflake private key.
+            role: Snowflake role.
+
+        There are multiple ways to authenticate with Snowflake. We are supporting the following
+        as of now:
+
+        1. externalbrowser
+        2. password
+        3. private_key
+        """
+
+        super().__init__(name, **properties)
+
+        params = {
+            "user": properties["user"],
+            "account": properties["account"],
+        }
+
+        if "authenticator" in properties:
+            params["authenticator"] = properties["authenticator"]
+
+        if "password" in properties:
+            params["password"] = properties["password"]
+
+        if "private_key" in properties:
+            params["private_key"] = properties["private_key"]
+
+        if "role" in properties:
+            params["role"] = properties["role"]
+
+        self.connection = SnowflakeConnection(**params)
+
+    def load_table(self, identifier: Union[str, Identifier]) -> Table:
+        sf_identifier = SnowflakeCatalog._SnowflakeIdentifier.table_from_string(
+            identifier if isinstance(identifier, str) else ".".join(identifier)
+        )
+
+        metadata_query = "SELECT SYSTEM$GET_ICEBERG_TABLE_INFORMATION(%s) AS METADATA"
+
+        with self.connection.cursor(DictCursor) as cursor:
+            try:
+                cursor.execute(metadata_query, (sf_identifier.table_name,))
+                metadata = json.loads(cursor.fetchone()["METADATA"])["metadataLocation"]
+            except Exception as e:
+                raise NoSuchTableError(f"Table {sf_identifier.table_name} not found") from e
+
+        _fs_scheme = urlparse(metadata)
+        _fs_props = {}
+
+        if _fs_scheme.scheme == "s3":
+            _fs_props.update(self._generate_s3_access_credentials())
+        elif _fs_scheme.scheme == "gcs":
+            assert os.environ.get(
+                "GOOGLE_APPLICATION_CREDENTIALS"
+            ), "GOOGLE_APPLICATION_CREDENTIALS not set. This is required for GCS access."
+        else:
+            warnings.warn(f"Unsupported filesystem scheme: {_fs_scheme.scheme}")
+
+        tbl = StaticTable.from_metadata(metadata, properties=_fs_props)
```

Comment:
Why do you return a `StaticTable` here instead of a full `Table`?

Reply:
Reading Snowflake generated `metadata.json` files may pose an issue. See #1106 for more details. Snowflake produces metadata files with some snapshot summary entries missing `operation` field.